core/schemas/dfiq.py

```diff
@@ -43,10 +44,21 @@ def load(cls, object: dict):
             return TYPE_MAPPING[object["type"]](**object)
         return cls(**object)
 
+    @classmethod
+    def from_yaml(cls, yaml_string: str) -> "DFIQBase":
+        try:
+            yaml_data = yaml.safe_load(yaml_string)
+        except yaml.YAMLError as e:
+            raise ValueError(f"Invalid YAML: {e}")
+        if yaml_data["type"] not in TYPE_MAPPING:
```

Comment:
Should you test if "type" not in yaml_data as well?

Reply:
Yes, I've added a check for this! Thanks